docs/README-develop-node-utils-APIs.md

```diff
@@ -3,32 +3,48 @@
 #### Table of Contents
 
 * [Overview](#overview)
-* [Start here: Clone the Repo](#clone)
+* [Before You Begin](#prerequisites)
+* [Start here: Fork and Clone the Repo](#clone)
 * [Basic Example: feature bash-shell](#simple)
  * [Step 1. YAML Definitions: feature bash-shell](#yaml)
  * [Step 2. Create the node_utils API: feature bash-shell](#api)
  * [Step 3. Create the Minitest: feature bash-shell](#minitest)
  * [Step 4. rubocop / lint: feature bash-shell](#lint)
+ * [Step 5. Build the gem](#gem)
 * [Advanced Example: router eigrp](#complex)
  * [Step 1. YAML Definitions: router eigrp](#comp_yaml)
  * [Step 2. Create the node_utils API: router eigrp](#comp_api)
  * [Step 3. Create the Minitest: router eigrp](#comp_minitest)
  * [Step 4. rubocop / lint: router eigrp](#comp_lint)
+ * [Step 5. Build the gem](#comp_gem)
 
 ## <a name="overview">Overview</a>
 
-This document is a HowTo guide for writing new cisco node_utils APIs. The node_utils APIs act as an interface between the NX-OS CLI and an agent's resource/provider. If written properly the new API will work as a common framework for multiple providers (Puppet, Chef, etc).
+This document is a HowTo guide for writing new cisco-network-node-utils APIs. The APIs act as an interface between the NX-OS CLI and an agent's resource/provider. If written properly the new API will work as a common framework for multiple providers (Puppet, Chef, etc).
 
-There are multiple components involved when creating new resources. This document focuses on the cisco node_utils API, command reference YAML files, and minitests.
+There are multiple components involved when creating new resources. This document focuses on the  cisco-network-node-utils API, command reference YAML files, and minitests.
 
 ![1](agent_files.png)
 
-## <a name="clone">Start here: Clone the Repo</a>
+## <a name="prerequisites">Before You Begin</a>
 
-First install the code base. Clone the cisco_node_utils repo into a workspace:
+This development guide uses tools that are packaged as gems that need to be installed on your workstation.
 
 ```bash
-git clone https://github.com/cisco/cisco-network-node-utils.git
+gem install cisco_nxapi
+gem install rubocop
+gem install minitest --version 4.3.2
+```
+
```

Comment:
Do we also need to walk people through installing git-flow? (https://github.com/petervanderdoes/gitflow-avh/wiki#installing-git-flow)

Also, if the training will have people working from an engineering server where they don't have admin/root privileges, you might want to include a note about using `gem install --user` and adding `~/.gem/ruby/x.x.x./bin` to their `PATH` - or you can move this step after the 'git clone' step and suggest using bundler instead to install the gems into the local repo copy (`bundle install --path ./vendor/bundle`)